src/stream/src/executor/agg_common.rs

```diff
@@ -44,15 +44,18 @@ pub struct AggExecutorArgs<S: StateStore> {
     pub watermark_epoch: AtomicU64Ref,
 
     // extra
-    pub extra: Option<AggExecutorArgsExtra>,
+    pub extra: E,
 }
 
-/// Extra arguments needed to construct an `XxxAggExecutor`.
-pub struct AggExecutorArgsExtra {
-    // hash agg specific things
-    pub group_key_indices: Vec<usize>,
+pub trait AggExecutorExtraArgs {}
 
-    // things only used by hash agg currently
-    pub metrics: Arc<StreamingMetrics>,
+pub struct SimpleAggExecutorExtraArgs {}
```

Comment:
```suggestion
pub struct SimpleAggExecutorExtraArgs;
```

Reply:
Actually I think empty struct is better than singleton struct in this case, considering future extension.